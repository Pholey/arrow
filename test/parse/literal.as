// Integer
20;
241_303;

0xdeadbeef;

0b01010101;

0o755;

// Real
5.12;
0.1;
5e+30;
4e10;
1e-2;

// String
"Hello World\n";

// Boolean
true;
false;
