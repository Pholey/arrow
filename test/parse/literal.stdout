{
  "tag": "Module",
  "span": "test/parse/literal.as:1-23",
  "statements": [
    {
      "tag": "Integer",
      "span": "test/parse/literal.as:2:1-3",
      "value": "20"
    },
    {
      "tag": "Integer",
      "span": "test/parse/literal.as:3:1-8",
      "value": "241303"
    },
    {
      "tag": "Integer",
      "span": "test/parse/literal.as:5:1-11",
      "value": "3735928559"
    },
    {
      "tag": "Integer",
      "span": "test/parse/literal.as:7:1-11",
      "value": "85"
    },
    {
      "tag": "Integer",
      "span": "test/parse/literal.as:9:1-6",
      "value": "493"
    },
    {
      "tag": "Float",
      "span": "test/parse/literal.as:12:1-5",
      "value": "5.120000"
    },
    {
      "tag": "Float",
      "span": "test/parse/literal.as:13:1-4",
      "value": "0.100000"
    },
    {
      "tag": "Float",
      "span": "test/parse/literal.as:14:1-6",
      "value": "4999999999999999999917321879552.000000"
    },
    {
      "tag": "Float",
      "span": "test/parse/literal.as:15:1-5",
      "value": "40000000000.000000"
    },
    {
      "tag": "Float",
      "span": "test/parse/literal.as:16:1-5",
      "value": "0.010000"
    },
    {
      "tag": "String",
      "span": "test/parse/literal.as:19:1-16",
      "value": "Hello World\n"
    },
    {
      "tag": "Boolean",
      "span": "test/parse/literal.as:22:1-5",
      "value": "true"
    },
    {
      "tag": "Boolean",
      "span": "test/parse/literal.as:23:1-6",
      "value": "false"
    }
  ]
}
